File: languages/java/src/Led.java
/**
 * Wrapper class for Flipper's on-board RGB LED.
 */
public class Led {

    private Flipper mFlipper;

    Led(Flipper flipper) {
        mFlipper = flipper;
    }

    /**
     * Configures the on-board RBG LED.
     */
    public void configure() {
        mFlipper.error.clear();
        mFlipper.getBinding().led_configure();
        Error.checkErrorCode(mFlipper.error.get());
    }

    /**
     * Sets the color of Flipper's RGB LED.
     * @param red The component of red, 0-255.
     * @param green The component of green, 0-255.
     * @param blue The component of blue, 0-255.
     */
    public void setRGB(int red, int green, int blue) {
        mFlipper.error.clear();
        mFlipper.getBinding().led_set_rgb((byte) red, (byte) green, (byte) blue);
        Error.checkErrorCode(mFlipper.error.get());
    }
}

File: languages/js/led.js
const ref = require('ref');

const led_interface = {
    // int configure();
    'configure': [ ref.types.uint32, [ ] ],

    // void rgb(uint8_t red, uint8_t green, uint8_t blue);
    'rgb': [ ref.types.void, [ ref.types.uint8, ref.types.uint8, ref.types.uint8 ] ],
};

function Led(flipper) {
    this.device = flipper;
    this.module = this.device.bind(led_interface, "led");
}

Led.prototype.configure = function() {
    return this.module.configure();
};

Led.prototype.rgb = function (red, green, blue) {
    return this.module.rgb(red, green, blue);
};

module.exports = Led;
